crates/client/eth/src/l1_messaging.rs

```diff
@@ -66,114 +68,83 @@ pub async fn sync(
-                        transaction hash on L2: {:?}",
-                        meta.block_number,
-                        meta.transaction_hash,
-                        meta.log_index,
-                        tx_hash
-                    );
-                }
-                Ok(None) => {}
-                Err(e) => {
-                    tracing::error!(
-                        "⟠ Unexpected error while processing L1 Message from block: {:?}, transaction_hash: {:?}, \
-                    log_index: {:?}, error: {:?}",
-                        meta.block_number,
-                        meta.transaction_hash,
-                        meta.log_index,
-                        e
-                    )
-                }
             }
+            Err(e) => tracing::error!("⟠ Unable to receive L1 -> L2 message event: {e}"),
         }
     }
 
     Ok(())
 }
 
-async fn process_l1_message(
+async fn process_l1_to_l2_msg(
     backend: &MadaraBackend,
-    event: &LogMessageToL2,
-    l1_block_number: &Option<u64>,
-    event_index: &Option<u64>,
+    client: &EthereumClient,
     chain_id: &ChainId,
-    mempool: Arc<Mempool>,
-) -> anyhow::Result<Option<Felt>> {
-    let transaction = parse_handle_l1_message_transaction(event)?;
-    let tx_nonce = transaction.nonce;
+    mempool: &Arc<Mempool>,
+    event: LogMessageToL2,
+    log: Log,
+) -> anyhow::Result<()> {
+    tracing::debug!("⟠ Processing L1 -> L2 message event {event:#?}, contract address: {}, block number: {:?}, transaction index: {:?}, transaction hash: {:?}, log index: {:?}",
+        log.address(), log.block_number, log.transaction_index, log.transaction_hash, log.log_index
+    );
 
+    let tx_nonce = Nonce(u256_to_felt(event.nonce)?);
     // Ensure that L1 message has not been executed
-    match backend.has_l1_messaging_nonce(tx_nonce) {
-        Ok(false) => {
-            backend.set_l1_messaging_nonce(tx_nonce)?;
-        }
-        Ok(true) => {
-            tracing::debug!("⟠ Event already processed: {:?}", transaction);
-            return Ok(None);
-        }
-        Err(e) => {
-            tracing::error!("⟠ Unexpected DB error: {:?}", e);
-            return Err(e.into());
-        }
-    };
+    if backend.has_l1_messaging_nonce(tx_nonce)? {
+        tracing::debug!("⟠ L1 -> L2 event already processed: {tx_nonce:?}");
+        return Ok(());
+    } else {
+        backend.set_l1_messaging_nonce(tx_nonce)?;
+    }
+
+    // Check if cancellation was initiated
+    let event_hash = get_l1_to_l2_msg_hash(&event);
+    let cancellation_timestamp = client.get_l1_to_l2_message_cancellations(event_hash).await?;
+    if cancellation_timestamp != Felt::ZERO {
+        tracing::info!("⟠ L1 message was cancelled at timestamp {:?}", cancellation_timestamp.to_biguint());
+        return Ok(());
+    }
+
+    let l1_handler_transaction = parse_handle_l1_message_transaction(&event)?;
```

Comment:
I wonder if it would be better to use our own primitive, mp_transaction::L1HandlerTransaction, even if it requires integrating starknet-api for blocking purposes.